src/state-mutation.js

```diff
@@ -18,8 +18,13 @@ const updateReverseRelationship = (
         const relCase = [1, 2]
           .map(i => pluralize(entity.get('type'), i))
           .find(r => foreignEntity.hasIn(['relationships', r]));
+        const relCount = foreignEntity.getIn(['relationships', relCase, 'data']);
+        const isset = relCount &&
+          relCount.getIn &&
+          relCount.getIn([0]) &&
+          relCount.getIn([0]).count();
```

Comment:
This seems to solve the issue with `relCount.getIn`, though I don't like a lot having so many conditions checked for a function that gets called a lot. 
On the other hand, it will bailout earlier on most of the cases, so it can be acceptable at this point.